test/e2e/kubectl.go

```diff
@@ -1201,3 +1200,34 @@ func streamingUpload(file *os.File, fileName string, postBodyWriter *multipart.W
 		Failf("Unable to close the writer for file upload. Error: %s", err)
 	}
 }
+
+var binPrefixes = []string{
+	"_output/dockerized/bin",
+	"_output/local/bin",
+	"platforms",
+}
+
+// findBinary searches through likely paths to find the specified binary.  It
+// takes the one that has been built most recently.
+func findBinary(binName string) (string, error) {
+	var binTime time.Time
+	var binPath string
+	var err error
+
+	for _, pre := range binPrefixes {
+		tryPath := path.Join(testContext.RepoRoot, pre, "linux/386", binName)
+		fi, err := os.Stat(tryPath)
+		if err != nil {
+			continue
+		}
+		if fi.ModTime().After(binTime) {
+			binPath = tryPath
+			binTime = fi.ModTime()
+		}
+	}
+
+	if len(binPath) > 0 {
+		return binPath, nil
+	}
+	return binPath, err
```

Comment:
rather than returning the most recent error (which looks like it comes from os.Stat many lines earlier), maybe just create a new error with content like "Binary [binName] not found in any of [binPrefixes]"?